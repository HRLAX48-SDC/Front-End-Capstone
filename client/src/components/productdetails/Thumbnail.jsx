import React from 'react';

function Thumbnail(props) {

  return (

    <div className='row'>
      {props.style.length ?
        props.style.map((itemStyle, index) => {
          return (
            <div key={index} className={index === props.styleNum ? 'p_thumbnail_selected col': 'p_thumbnail col'}>
              <div className='col-sd-3'>
                <img
                  className='p_thumbnail_img'
                  src={itemStyle.photos[0].thumbnail_url}
                  onClick={() => props.handleClick(index)}>
                </img>
                <div className='p_thumbnail_font'>{itemStyle.name}</div>
              </div>
            </div>
          )
        })
      :
        <div className='spinner-border' role='status'>
          <span className='visually-hidden'>Loading...</span>
        </div>
      }
    </div>
  )
}

export default Thumbnail;