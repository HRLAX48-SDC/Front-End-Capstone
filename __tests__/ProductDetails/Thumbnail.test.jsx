import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import Thumbnail from '../../client/src/components/productdetails/Thumbnail.jsx';

test('renders Thumbnail component correctly', () => {
  const style = [{
    style_id: 253620,
    name: 'Forest Green & Black',
    original_price: '140.00',
    sale_price: null,
    photos: [{
      thumbnail_url: "https://images.unsplash.com/photo-1501088430049-71c79fa3283e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80",
      url: "https://images.unsplash.com/photo-1501088430049-71c79fa3283e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=668&q=80"
    }]
  }];
  const styleNum = 0;
  const thumbnail = renderer.create(<Thumbnail style={style} styleNum={styleNum} />).toJSON();
  expect(thumbnail).toMatchSnapshot();
});