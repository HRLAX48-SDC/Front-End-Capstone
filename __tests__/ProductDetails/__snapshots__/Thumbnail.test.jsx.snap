// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Thumbnail component correctly 1`] = `
<div
  className="p_thumbnail_container"
>
  <div
    className="p_thumbnail_selected p_thumbnail_child"
  >
    <i
      className="fas fa-check-circle p_check"
    />
    <img
      className="p_thumbnail_img"
      onClick={[Function]}
      src="https://images.unsplash.com/photo-1501088430049-71c79fa3283e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
    />
    <div
      className="p_thumbnail_font"
    >
      Forest Green & Black
    </div>
  </div>
</div>
`;
